docs(linter): Update docs for no-optional-chaining rule to clarify its safety today

connorshea opened the pull request:
The rule is safe to use nowadays, as all major browsers have supported it for over 5 years.

camc314 commented:
### Merge activity

* **Dec 11, 9:17 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Dec 11, 9:17 AM UTC**: `camc314` added this pull request to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).

connorshea committed:
docs(linter): Update docs for no-optional-chaining rule to clarify its safety today (#16727)

The rule is safe to use nowadays, as all major browsers have supported it for over 5 years.
